server/helpers/adapters_tracker.go

```diff
@@ -171,6 +265,28 @@ func (a *AdaptersTracker) UndeployAdapter(ctx context.Context, adapter models.Ad
 			return ErrAdapterAdministration(err)
 		}
 
+	case "kubernetes":
+		kubeClient, err := meshkitkube.New([]byte(""))
+		if err != nil {
+			return ErrAdapterAdministration(err)
+		}
+
+		if adapter.Name == "" {
+			adapter = a.adapters[adapter.Location]
+		}
+
+		// Delete the Deployment
+		err = kubeClient.KubeClient.AppsV1().Deployments(core.MesheryNamespace).Delete(context.Background(), adapter.Name, metav1.DeleteOptions{})
+		if err != nil {
+			return ErrAdapterAdministration(err)
+		}
+
+		// Delete the associated Service objects
```

Comment:
The collective sequence that we have here is a bit imperative. Ideally, we are simply reusing Helm client to hand Kubernetes a manifest (a chart) and Helm/Kubernetes does the rest.

Reply:
I've update the code to use Helm client. I don't think this way the adapter is completely decoupled from meshery.